core/vmpu/src/mpu_armv8m/vmpu_armv8m_unpriv_access.c

```diff
@@ -115,12 +49,13 @@ uint32_t vmpu_unpriv_access(uint32_t addr, uint32_t size, uint32_t data)
                 default:
                     break;
             }
-            break;
         }
+
         if (++tries > 1 || !vmpu_fault_recovery_mpu(0, 0, addr, 0)) {
```

Comment:
We should have to recover. `vmpu_buffer_access_is_ok` should tell you what access you would have if you did recover. The access can happen from secure mode without the recovery happening.